src/query-builder/SelectQueryBuilder.ts

```diff
@@ -3750,8 +3769,17 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
 
     /**
      * Loads raw results from the database.
+     * @param {Date} [timestamp] A timestamp to get datasets at a specific time. If undefined use a date in the future to get the current datasets.
      */
-    protected async loadRawResults(queryRunner: QueryRunner) {
+    protected async loadRawResults(
+        queryRunner: QueryRunner,
+        timestamp = dayjs().add(10, "year").toDate(),
```

Comment:
why exactly 10 years? Better to remove default and let user decide the defaults.